dashboard/app/models/pd/workshop.rb

```diff
@@ -580,6 +580,34 @@ def attending_teachers
     sessions.flat_map(&:attendances).flat_map(&:teacher).uniq
   end
 
+  # Get all teachers who have attended all sessions of this workshop.
+  def teachers_attending_all_sessions(cdo_scholarship=false)
```

Comment:
I might rename this parameter to something like filter_on_scholarship so it's clear false gives you all the teachers and true gives you just those with scholarships. The current name could be interpreted as false=teachers without scholarships

Reply:
Good idea, thanks!